### PR TITLE
Integrate V4 QB

### PR DESCRIPTION
**Pull request by nityanandagohain**
Follow up of https://github.com/SigNoz/signoz/pull/5912 . This PR contains changes of https://github.com/SigNoz/signoz/pull/5912 as well. So please review it first before coming to this PR.

For issue https://github.com/SigNoz/signoz/issues/5555

Here the different API's are configured to use the new schema in a feature flag.

*  GetLogFields() use the new table if flag is enabled.
* UpdateLogField() uses the new table if flag is enabled ( selected field api)
*  GetLogAggregateAttributes() uses the new table if flag is enabled.
*  GetLogAttributeKeys() uses the new table if flag is enabled.
*  GetLogAttributeValues() uses the new table if flag is enabled.
* LiveTailLogsV4() is used if flag is enabled else LiveTailV3
* Query Builder options is moved to model.

### CODE REVIEW

**Comment by srikanthccv** [comment]
Whereever there are big functions, can you make a v2 version of it and call it at the start of call chain instead of `if use { a lot new schema code} else {lot of old code}`

- It will be easier to follow the changes in PR because you won't touch old code
- it will be easier to delete when the v2 rollout is complete

**Comment by nityanandagohain** [comment]
Thanks for the feedback, it makes sense. Let me do that.

File: pkg/query-service/utils/format.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ import (
 
 // ValidateAndCastValue validates and casts the value of a key to the corresponding data type of the key
 func ValidateAndCastValue(v interface{}, dataType v3.AttributeKeyDataType) (interface{}, error) {
+	// get the actual value if it's a pointer
+	v = getPointerValue(v)
```
Comment (srikanthccv) [inline pkg/query-service/utils/format.go:18]: Why is this added?

Comment (nityanandagohain) [reply pkg/query-service/utils/format.go:18]: Added it due to this, so in the new list api that we are using, we add the id right, which I get as a pointer to a string.
https://github.com/SigNoz/signoz/blob/3fdfb51e0243d4c3587204c65152d71b9b2ea0df/pkg/query-service/app/querier/querier.go#L517
 
image

File: pkg/query-service/app/clickhouseReader/reader.go [hunk 2]
```diff
@@ -5262,6 +5355,59 @@ func (r *ClickHouseReader) GetSpanAttributeKeys(ctx context.Context) (map[string
...
+			zap.L().Debug("closing go routine : " + client.Name)
+			return
+		case <-ticker.C:
+			// get the new 100 logs as anything more older won't make sense
+			var tmpQuery string
+			bucketStart := (timestampStart / NANOSECOND) - 1800
+
+			// we have to form the query differently if the resource filters are used
+			if strings.Contains(query, r.logsResourceTableV2) {
+				tmpQuery = fmt.Sprintf("seen_at_ts_bucket_start >=%d)) AND ts_bucket_start >=%d AND timestamp >=%d", bucketStart, bucketStart, timestampStart)
```
[the remark this answers is not in the document]
Comment (nityanandagohain) [reply pkg/query-service/app/clickhouseReader/reader.go:5381]: the other part is the resource filter which is present in the query builder.

[the remark this answers is not in the document]
Comment (nityanandagohain) [reply pkg/query-service/app/clickhouseReader/reader.go:5381]: the parenthesis is like this because when the resource table is present it means that there is a subquery. So  i am adding `seen_at_ts_bucket_start` with closing parenthesis here as it needs to be updated.


The cleaner way was to use String formatting but that did work when there were where clauses with `%`. Eg: attr contains '%select%'. It broke the formatting.

[the remark this answers is not in the document]
Comment (srikanthccv) [reply pkg/query-service/app/clickhouseReader/reader.go:5381]: ok

**Review by srikanthccv — DISMISSED** [review]

**Review by srikanthccv — APPROVED** [review]
